python/lsst/daf/persistence/swiftStorage.py

```diff
@@ -0,0 +1,519 @@
+        try:
+            self._connection.head_container(self._containerName)
+        except self.swift.ClientException:
+            return False
+        return True
+
+    def deleteContainer(self):
+        """Delete all the objects in this container"""
+        respHeaders, containers = self._connection.get_account()
+        for container in containers:
+            name = container['name']
+            if name == self._containerName:
+                headers, objects = self._connection.get_container(
+                    self._containerName)
+                for obj in objects:
+                    self._connection.delete_object(self._containerName, obj['name'])
+                self._connection.delete_container(self._containerName)
+                break
+
+    def write(self, butlerLocation, obj):
+        """Writes an object to a location and persistence format specified by
+        ButlerLocation
+
+        This file uses PosixStorage to write the object to a file on disk (that
+        is to say: serialize it). Then the file is uploaded to the swift
+        container. When we have better support for pluggable serializers,
+        hopefully the first step of writing to disk can be skipped and the
+        object can be serialzied and streamed directly to the swift container.
+
+        Parameters
+        ----------
+        butlerLocation : ButlerLocation
+            The location & formatting for the object to be written.
+        obj : object instance
+            The object to be written.
+        """
+        swiftLocations = butlerLocation.getLocations()
+        # Here the ButlerLocation is modified sligtly to write to a temporary
+        # file via PosixStorage. (Then the temporary file is written to the
+        # swift container)
+        localFile = tempfile.NamedTemporaryFile()
+        butlerLocation.locationList = [localFile.name]
+        butlerLocation.storage = PosixStorage('/')
+        butlerLocation.storage.write(butlerLocation, obj)
+        butlerLocation.locationList = swiftLocations
+        with open(localFile.name, 'rb') as f:
+            self._connection.put_object(
+                container=self._containerName, obj=swiftLocations[0],
+                contents=f, content_length=os.stat(localFile.name).st_size)
+
+    def read(self, butlerLocation):
+        """Read from a butlerLocation.
+
+        Parameters
+        ----------
+        butlerLocation : ButlerLocation
+            The location & formatting for the object(s) to be read.
+
+        Returns
+        -------
+        A list of objects as described by the butler location. One item for
+        each location in butlerLocation.getLocations()
+        """
+        localFile = self.getLocalFile(butlerLocation.getLocations()[0])
+        butlerLocation.locationList = [localFile.name]
+        butlerLocation.storage = PosixStorage('/')
+        obj = butlerLocation.storage.read(butlerLocation)
+        return obj
```

Comment:
The implementation is not consistent with docstrings which says that a list is returned and it includes all locations, not just first one?